src/message.rs

```diff
@@ -164,3 +164,16 @@ where
         (**self).clear()
     }
 }
+
+/// An unknown field on a protobuf message.
+///
+/// Structs generated by `prost-build` with the `unknown_fields` flag enabled have a field
+/// `protobuf_unknown_fields: Vec<UnknownField>` that gives access to any fields that were not
+/// recognized.
+/// Any values added to this field will also be present in serialized form.
+#[derive(Debug, Default, Clone, PartialEq)]
+pub struct UnknownField {
+    pub tag: u32,
+    /// The raw field value.
+    pub value: Vec<u8>,
```

Comment:
The C++ implementation handles unknown fields differently. It actually parses the fields. Why is this approach chosen? Can you explain the advantages of both approaches? https://protobuf.dev/reference/cpp/api-docs/google.protobuf.unknown_field_set/#UnknownField